packages/plugins/Traderxyz/src/SNSAdaptor/PostPreview.tsx

```diff
@@ -0,0 +1,304 @@
+    },
+
+    previewBoxInnerGridContainerItem: {
+        display: 'table-cell',
+        height: '190px',
+        border: '0px solid green',
+    },
+    previewBoxInnerGridContainerItemImg: {
+        height: '100%',
+        maxWidth: '100%',
+        objectFit: 'cover',
+        borderRadius: '10px',
+        userSelect: 'none',
+    },
+    previewBoxInnerGridContainerChip: {
+        position: 'relative',
+        left: '65%',
+        top: '-35px',
+        background: theme.palette.mode === 'dark' ? '#1D2933' : 'rgb(23, 23, 23)',
+        color: 'white',
+        fontSize: '18px',
+        border: '1px solid rgb(0, 0, 0)',
+        padding: '10px 5px 10px',
+    },
+    previewBoxInnerGridFooterTitle1: {
+        fontWeight: 500,
+        fontSize: '18px',
+        color: theme.palette.mode === 'dark' ? '#ffffff' : '#000000',
+    },
+    previewBoxInnerGridFooterTitle2: {
+        fontWeight: 'normal',
+        fontSize: '14px',
+        color: theme.palette.mode === 'dark' ? '#ffffff' : '#000000',
+    },
+}))
+
+/**
+ * This page we use to full fill order
+ */
+
+export function PostPreview({ info }: { info: TradeMetaData }) {
+    const selectedChainId = useChainId()
+    const nftSwapSdk = useTraderApi(selectedChainId)
+    const account = useAccount()
+    const { classes } = useStyles()
+    const t = useI18N()
+    const theme = useTheme()
+    const { showSnackbar } = useCustomSnackbar()
+
+    const signOrder = useCallback(async () => {
+        const normalizeSignedOrder = nftSwapSdk.normalizeSignedOrder(info.signedOrder)
+        const walletAddressUserA = account
+        const assetsToSwapUserA = [info.assetsInfo.receiving_token, ...info.assetsInfo.nfts]
+
+        /// Check order status
+
+        // Invalid = 0,
+        // InvalidMakerAssetAmount = 1,
+        // InvalidTakerAssetAmount = 2,
+        // Fillable = 3,
+        // Expired = 4,
+        // FullyFilled = 5,
+        // Cancelled = 6
+
+        const orderStatus = await nftSwapSdk.getOrderStatus(normalizeSignedOrder)
+
+        if (orderStatus === 3) {
+            // Fillable
+            // Check if we need to approve the NFT for swapping
+            const approvalStatusForUserA = await nftSwapSdk.loadApprovalStatus(
+                assetsToSwapUserA[0] as SwappableAsset,
+                walletAddressUserA,
+            )
+
+            // Check if buyer needs to approve the sell token contract or no
+            if (!approvalStatusForUserA.contractApproved) {
+                // If not approved
+                const approvalTx = await nftSwapSdk.approveTokenOrNftByAsset(
+                    assetsToSwapUserA[0] as SwappableAsset,
+                    walletAddressUserA,
+                )
+
+                await approvalTx.wait().then(
+                    (msg) =>
+                        showSnackbar(
+                            `Approved ${assetsToSwapUserA[0]?.tokenAddress} contract to swap with 0x (txHash: ${msg.transactionHash})`,
```

Comment:
plz extract the template string as variable.

Reply:
Fixed.